tests/system/test_dbapi.py

```diff
@@ -59,779 +57,757 @@ def raw_database(shared_instance, database_operation_timeout, not_postgres):
-        """
-        )
-
-        # Insert data to table
-        cur.execute(
-            sql="INSERT INTO JsonDetails (DataId, Details) VALUES (%s, %s)",
-            args=(123, JsonObject({"name": "Jakob", "age": "26"})),
-        )
+            )
 
-        # Read back the data.
-        cur.execute("""select * from JsonDetails;""")
-        got_rows = cur.fetchall()
+        self._cursor.execute("SELECT * FROM contacts")
+        self._conn.commit()
 
-        # Assert the response
-        assert len(got_rows) == 1
-        assert got_rows[0][0] == 123
-        assert got_rows[0][1] == {"age": "26", "name": "Jakob"}
+    def test_staleness(self):
+        """Check the DB API `staleness` option."""
 
-        # Drop the table
-        cur.execute("DROP TABLE JsonDetails")
-        conn.commit()
-        conn.close()
-    finally:
-        # Delete table
-        table = dbapi_database.table("JsonDetails")
-        if table.exists():
-            op = dbapi_database.update_ddl(["DROP TABLE JsonDetails"])
-            op.result()
-
-
-@pytest.mark.skipif(_helpers.USE_EMULATOR, reason="Emulator does not support json.")
-def test_json_array(shared_instance, dbapi_database):
-    try:
-        # Create table
-        conn = Connection(shared_instance, dbapi_database)
-        conn.autocommit = True
+        before_insert = datetime.datetime.utcnow().replace(tzinfo=UTC)
+        time.sleep(0.25)
 
-        cur = conn.cursor()
-        cur.execute(
+        self._cursor.execute(
             """
-            CREATE TABLE JsonDetails (
-                DataId     INT64 NOT NULL,
-                Details    JSON,
-            ) PRIMARY KEY (DataId)
+    INSERT INTO contacts (contact_id, first_name, last_name, email)
+    VALUES (1, 'first-name', 'last-name', 'test.email@example.com')
         """
         )
-        cur.execute(
-            "INSERT INTO JsonDetails (DataId, Details) VALUES (%s, %s)",
-            [1, JsonObject([1, 2, 3])],
-        )
-
-        cur.execute("SELECT * FROM JsonDetails WHERE DataId = 1")
-        row = cur.fetchone()
-        assert isinstance(row[1], JsonObject)
-        assert row[1].serialize() == "[1,2,3]"
-
-        cur.execute("DROP TABLE JsonDetails")
-        conn.close()
-    finally:
-        # Delete table
-        table = dbapi_database.table("JsonDetails")
-        if table.exists():
-            op = dbapi_database.update_ddl(["DROP TABLE JsonDetails"])
-            op.result()
-
-
-def test_DDL_commit(shared_instance, dbapi_database):
-    """Check that DDLs in commit mode are executed on calling `commit()`."""
-    try:
-        conn = Connection(shared_instance, dbapi_database)
-        cur = conn.cursor()
-
-        cur.execute(
+        self._conn.commit()
+
+        self._conn.read_only = True
+        self._conn.staleness = {"read_timestamp": before_insert}
+        self._cursor.execute("SELECT * FROM contacts")
+        self._conn.commit()
+        assert len(self._cursor.fetchall()) == 0
```

Comment:
Agree. Just to be clear its the existing test not related to this PR.

IMO there should definitely be separate tests in case they were unit tests but probably ok for system tests?